frontend/src/component/insights/componentsChart/CreationArchiveChart/CreationArchiveChart.tsx

```diff
@@ -35,93 +65,240 @@ type RawWeekData = {
+                return creationVsArchivedChart.datasets
                     .map((d) => d.data.find((item) => item.week === label))
-                    .reduce(
-                        (acc: WeekData, item: RawWeekData) => {
-                            if (item) {
-                                acc.archivedFlags += item.archivedFlags || 0;
-                                const createdFlagsSum = item.createdFlags
-                                    ? Object.values(item.createdFlags).reduce(
-                                          (sum: number, count: number) =>
-                                              sum + count,
-                                          0,
-                                      )
-                                    : 0;
-                                acc.totalCreatedFlags += createdFlagsSum;
-                            }
-                            if (!acc.date) {
-                                acc.date = item?.date;
-                            }
-                            return acc;
-                        },
-                        {
-                            archivedFlags: 0,
-                            totalCreatedFlags: 0,
-                            week: label,
-                        } as WeekData,
-                    );
+                    .reduce(aggregateWeekData, createInitialWeekData(label));
             })
+            .map((week) => ({
+                ...week,
+                archivePercentage:
+                    week.totalCreatedFlags > 0
+                        ? (week.archivedFlags / week.totalCreatedFlags) * 100
+                        : 0,
+            }))
             .sort((a, b) => (a.week > b.week ? 1 : -1));
+
+        const flagTypeColors = getFlagTypeColors(theme);
+
+        const flagTypeDatasets = Array.from(allFlagTypes).map(
+            (flagType, index) => ({
+                label: flagType,
+                data: weeks,
+                backgroundColor: flagTypeColors[index % flagTypeColors.length],
+                borderColor: flagTypeColors[index % flagTypeColors.length],
+                type: 'bar' as const,
+                parsing: {
+                    yAxisKey: `createdFlagsByType.${flagType}`,
+                    xAxisKey: 'date',
+                },
+                yAxisID: 'y',
+                stack: 'created',
+                order: 2,
+            }),
+        );
+
+        const flagTypeNames = Array.from(allFlagTypes);
+
         return {
             datasets: [
                 {
-                    label: 'Number of created flags',
+                    label: 'Archived flags',
+                    data: weeks,
+                    backgroundColor: theme.palette.background.application,
+                    borderColor: theme.palette.background.application,
+                    parsing: { yAxisKey: 'archivedFlags', xAxisKey: 'date' },
+                    order: 2,
+                },
+                ...flagTypeDatasets,
```

Comment:
Each green bar on bar chart is another dataset. So total there is over 5 datasets.